azurerm/internal/services/maintenance/resource_arm_maintenance_configuration.go

```diff
@@ -0,0 +1,168 @@
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMaintenanceConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMaintenanceConfigurationCreateUpdate,
+		Read:   resourceArmMaintenanceConfigurationRead,
+		Update: resourceArmMaintenanceConfigurationCreateUpdate,
+		Delete: resourceArmMaintenanceConfigurationDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.MaintenanceConfigurationID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"scope": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "All",
+				ValidateFunc: validation.StringInSlice([]string{
+					"All",
+					"Host",
+					"Resource",
+					"InResource",
+				}, false),
+			},
+
+			// can not contain upper case key
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmMaintenanceConfigurationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Maintenance.ConfigurationsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing MaintenanceConfiguration %q (Resource Group %q): %+v", name, resGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_maintenance_configuration", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	scope := d.Get("scope").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	configuration := maintenance.Configuration{
+		Name:     utils.String(name),
+		Location: utils.String(location),
+		ConfigurationProperties: &maintenance.ConfigurationProperties{
+			MaintenanceScope: maintenance.Scope(scope),
+			Namespace:        utils.String("Microsoft.Maintenance"),
+		},
+		Tags: tags.Expand(t),
```

Comment:
We can just get the values inline
```suggestion
		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
```